membership-attribute-service/app/services/ZuoraAttributeService.scala

```diff
@@ -0,0 +1,141 @@
+
+    def sub(accountId: AccountId): Future[Disjunction[String, List[Subscription[AnyPlan]]]] = subscriptionsForAccountId(accountId)(anyPlanReads)
+
+    val maybeSubs: Future[Disjunction[String, List[Subscription[AnyPlan]]]] = accountIds.traverse[Future, Disjunction[String, List[Subscription[AnyPlan]]]](id => {
+      sub(id)
+    }).map(_.sequenceU.map(_.flatten))
+
+    maybeSubs.map {
+      _.leftMap { errorMsg =>
+        log.warn(s"We tried getting subscription for a user with identityId $identityId, but then $errorMsg")
+        s"We called Zuora to get subscriptions for a user with identityId $identityId but the call failed because $errorMsg"
+      } map { subs =>
+        log.info(s"We got subs for identityId $identityId from Zuora and there were ${subs.length}")
+        subs
+      }
+    }
+  }
+
+  private def withTimer[R](whichCall: String, futureResult: Future[Disjunction[String, R]], identityId: String) = {
+    import loghandling.StopWatch
+    val stopWatch = new StopWatch
+
+    futureResult.map { disjunction: Disjunction[String, R] =>
+      disjunction match {
+        case -\/(message) => log.warn(s"$whichCall failed with: $message")
+        case \/-(_) =>
+          val latency = stopWatch.elapsed
+          val zuoraConcurrencyCount = ZuoraRequestCounter.get
+          val customFields: List[LogField] = List("zuora_latency_millis" -> latency.toInt, "zuora_call" -> whichCall, "identityId" -> identityId, "zuora_concurrency_count" -> zuoraConcurrencyCount)
+          logInfoWithCustomFields(s"$whichCall took ${latency}ms.", customFields)
+      }
+    }.onFailure {
+      case e: Throwable => log.error(s"Future failed when attempting $whichCall.", e)
+    }
+    futureResult
+  }
+
+  def dynamoAndZuoraAgree(attributesFromDynamo: Future[Option[Attributes]], attributesFromZuora: Future[Option[Attributes]], identityId: String): Future[Boolean] = {
+    attributesFromDynamo flatMap { maybeDynamoAttributes =>
+      attributesFromZuora map { maybeZuoraAttributes =>
+        val zuoraAttributesWithIgnoredFields = maybeZuoraAttributes flatMap  { zuoraAttributes =>
+          maybeDynamoAttributes map { dynamoAttributes =>
+            zuoraAttributes.copy(
+              AdFree = dynamoAttributes.AdFree, //fetched from Dynamo in the Zuora lookup anyway (dynamo is the source of truth)
+              Wallet = dynamoAttributes.Wallet, //can't be found based on Zuora lookups, and not currently used
+              MembershipNumber = dynamoAttributes.MembershipNumber, //I don't think membership number is needed and it comes from Salesforce
+              MembershipJoinDate = dynamoAttributes.MembershipJoinDate.flatMap(_ => zuoraAttributes.MembershipJoinDate), //only compare if dynamo has value
+              DigitalSubscriptionExpiryDate = None
+            )
+          }
+        }
+        val dynamoAndZuoraAgree = zuoraAttributesWithIgnoredFields == maybeDynamoAttributes
```

Comment:
interesting approach, I'd have probably done it by pattern matching the relevant fields from both and then just comparing them either in a tuple or just individually, otherwise you're comparing fields that you just set to the same.  However since it's hopefully temporary code it's not worth going into detail.